sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -649,6 +662,11 @@ public partial class ResponseClassifier
         public bool TryGetValue(string name, out string? value) { throw null; }
         public bool TryGetValues(string name, out System.Collections.Generic.IEnumerable<string>? values) { throw null; }
     }
+    public abstract partial class RetryCondition
```

Comment:
Why do we need this type? Shouldn't the RetryPolicy be doing reties and so it knows then to retry?

Reply:
The goal of making this suggestion in this Draft Discussion PR, was to suggest an approach to introducing the "composition" feature without needing multiple additional policies in the pipeline.  There is a discussion regarding this pattern here: https://github.com/Azure/azure-sdk-for-net/pull/31892#discussion_r999782115

Reply:
I think we have too many places that can make retry decisions already.

Reply:
Sounds good - thanks for considering my ideas, and excited to see your implementation, @JoshLove-msft!